src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
```

Comment:
https://github.com/github/hubot/commit/20fce05297bf821063863307433f3fb5a7f1926a adds a default `done`, and I like it better. It will fix #1021 transparently.